### ISSUE TITLE
False Positive in "The result of the '<<' expression is undefined"

### ISSUE THREAD

**Issue by hiraditya** [llvm/llvm-project#18008]
|  |  |
| --- | --- |
| Bugzilla Link | [17634](https://llvm.org/bz17634) |
| Version | unspecified |
| OS | Linux |
| CC | @AnnaZaks,@hiraditya,@belkadan |

## Extended Description 
For the following program I get the report 
Location:	line 9, column 11
Description:	The result of the '<<' expression is undefined

```cpp
int foo(int a, int b) {
int x, s;
int z = 0;
while (b<=a) {
  s = 0;
  while ((b<<s) <= a) {
    s++;
  }
  x = 1 << (s-1); // Line 9
  z += x;
  }
return z;
}
```

**Comment by hiraditya** [comment]
can't repro anymore with clang version 18.0.0

**Comment by Endilll** [comment]
@hiraditya Can you provide a Compiler Explorer link?

**Comment by hiraditya** [comment]
> @hiraditya Can you provide a Compiler Explorer link?

https://godbolt.org/z/eY6Pac7aT
